server/axios_api.js

```diff
@@ -0,0 +1,62 @@
+var axios = require("axios").default;
+var HOST = "movies-tvshows-data-imdb.p.rapidapi.com";
+var API_KEY = "030611f594msh73349b7b00af6e8p16d57ejsncbf1b45a6165";
+var URL = "https://movies-tvshows-data-imdb.p.rapidapi.com/";
+
+
+
+function get_details(id) {
+  var options = {
+    method: 'GET',
+    url: URL,
+    params: {type: 'get-movie-details', imdb: id},
+    headers: {
+      'x-rapidapi-host': HOST,
+      'x-rapidapi-key': API_KEY
+    }
+  };
+  axios.request(options).then(function (response) {
+    console.log("rating:"+response.data["imdb_rating"]+"\n"+"popularity:"+response.data["popularity"]+"\n"+"vote:"+response.data["vote_count"]);
+  }).catch(function (error) {
+    console.error(error);
+  });
+
+}
+
+
+function get_id(t){
+  var options = {
+    method: 'GET',
+    url: URL,
+    params: {type: 'get-movies-by-title', title: t},
+    headers: {
+      'x-rapidapi-host': HOST,
+      'x-rapidapi-key': API_KEY
+    }
+  };
+  
+  axios.request(options).then(function (response) {
+    var results = response.data.movie_results;
+    results.forEach(function(e,i){
+      console.log(e['imdb_id'])
+      return(e['imdb_id'])
+    });
+  }).catch(function (error) {
+    console.error(error);
+  });
+}
+
+
+
+// Local test
+
+async function main(){
+  var test_query = "i am legend";
+  var id = await get_id(test_query);
```

Comment:
does this API support get for all details just by movie title? Wondering if we can avoid 2 API calls for each movie then it will be great.

Reply:
@isha-bansal0115 So far, I haven't found an api call that can handle returning multiple movies on a single api call.

Reply:
@lgdeloss  May be I should add more context.

What I was saying is that in this flow, for a given title I need to first get ID and then its details right?
Does this API give details if we just give title? Why do we have to get movie ID first?

Reply:
@isha-bansal0115 Oh I see what you're asking. From what I gathered, we need to first get that imdb_id and then use that id to query the details on the movie

Reply:
@lgdeloss if thats the case, can you please explore OMDB API which I have added. That returns with just title. If that makes sense, then we can change API?

2 API calls are risky in sense that if first fails then we are stuck. Or can you check if this API can return with just title?

Reply:
@isha-bansal0115 ok, I'm looking at the api you found now, the api you found definitely is more concise! Nice find!